### PR TITLE
Create TimedeltaAttribute

### PR DESCRIPTION
**Pull request by mxr**

### CODE REVIEW

File: pynamodb_attributes/timedelta.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from typing import Optional
+
+import pynamodb
+
+from ._typing import Attribute
+
+
+class TimedeltaAttribute(Attribute[timedelta]):
+    """"
+    Stores a time range as a Python timedelta (truncated to seconds) in a DynamoDB number.
```
Comment (ikonst) [inline pynamodb_attributes/timedelta.py]: "time duration" perhaps?

**Review by ikonst — DISMISSED** [review]

File: pynamodb_attributes/timedelta.py (hunk 1 above)
Comment (mxr) [reply pynamodb_attributes/timedelta.py]: Sure

**Comment by mxr** [comment]
@ikonst should we validate the input value more? For example, `timedelta(seconds=1.5)` or `timedelta(milliseconds=1)` can't be accurately represented by `TimedeltaAttribute` because of lossiness during serialization

**Comment by ikonst** [comment]
What if we just don't truncate to integer?

**Comment by mxr** [comment]
Sure that could work or we could store as truncated microseconds only? Since microsecond precision is the best you can get from timedelta anyway

**Comment by ikonst** [comment]
You mean to offer only TimedeltaUs?

**Comment by mxr** [comment]
Yeah I guess all 3 offer the same user API but have 3 different serialization formats under the hood depending on the desired precision. Which on 2nd thought seems weird

**Comment by ikonst** [comment]
I guess it can matter how you represent data internally since all kinds of stuff access it directly (through dynamostreams etc.).

**Comment by mxr** [comment]
But we can't possibly represent all possible formats (int, float, sec, ms, us)...?

**Comment by mxr** [comment]
I could just model this after the datetime attribute and model the internals  as sec, ms, or us, and have people accept that truncation will be inevitable here as it is there

File: pynamodb_attributes/timedelta.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from typing import Optional
+
+import pynamodb
+
+from ._typing import Attribute
+
+
+class TimedeltaAttribute(Attribute[timedelta]):
+    """"
+    Stores a time duration as a Python timedelta (truncated to seconds) in a DynamoDB number.
```
Comment (ikonst) [inline pynamodb_attributes/timedelta.py]: "Stores ... as a Python timedelta" doesn't seem right :)

Maybe simply
```suggestion
    Stores a timedelta as a number of seconds (truncated).
```
?

Comment (mxr) [reply pynamodb_attributes/timedelta.py]: Sure

**Review by ikonst — DISMISSED** [review]

**Review by ikonst — APPROVED** [review]

**Pull request merged**